pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -410,22 +435,50 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 	}
 
 	// Update the pull secret object if required
-	modifiedCD, err := r.updatePullSecretInfo(pullSecret, cd, cdLog)
-	if err != nil || modifiedCD {
-		if err != nil {
-			cdLog.WithError(err).Error("Error updating the merged pull secret")
-			return reconcile.Result{}, err
-		}
+	switch modifiedCD, err := r.updatePullSecretInfo(pullSecret, cd, cdLog); {
+	case err != nil:
+		cdLog.WithError(err).Error("Error updating the merged pull secret")
+		return reconcile.Result{}, err
+	case modifiedCD:
```

Comment:
This is not related to cluster provision change, so if possible it should be a separate commit

Reply:
I will remove this. It is a remnant of when I changed things to create the pull secret anew with every ClusterProvision, not realizing at the time that the pull secret was needed for the image resolution job as well.